.special-jobs.yml

```diff
@@ -0,0 +1,43 @@
+---
+stages:
+  - .pre
+  - lint
+  - notify
+  - deploy_containers
+
+jobs:
```

Comment:
It took me some time to understand why we want to have these jobs in a special list. But if I understand correctly, the end goal of this linter is to make sure that all the jobs we add define both `rules` and `needs`. I think it would be clearer if this field had a more explicit name, like `needs_rules_allowlist` or even two different lists with `no_rules_allowlist` and `no_needs_allowlist`.
Because I feel like these jobs are not really "special" but they are rather here because in the current state they miss/we do not want them to have  either a need or a rules